components/public-api-server/pkg/proxy/prometheusmetrics.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package proxy
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"time"
+)
+
+func ReportConnectionDuration(d time.Duration) {
+	ConnectionDurationHistogram.Observe(d.Seconds())
+}
+
+var ConnectionDurationHistogram = prometheus.NewHistogram(prometheus.HistogramOpts{
+	Namespace: "gitpod",
+	Name:      "grpc_connection_duration_seconds",
```

Comment:
```suggestion
	Name:      "public_api_proxy_connection_create_duration_seconds",
```
Metric should be specific, and should include units where relevant. See https://prometheus.io/docs/practices/naming/